fruitCollection.js

```diff
@@ -0,0 +1,115 @@
+/*
+monkey eating fruit
+tree to tree
+find collection of tree of fruit
+
+list of tree
+each tree: x-coordinate, y-coordinate, num of fruit, vine length
+
+i: [2,4,5,10]
+o: max of fruit 
+
+const trees = [
+    [2, 4, 2, 6],
+    [10, 4, 5, 5],
+    [3, 8, 3, 6],
+    [12, 8, 6, 5],
+    ,[1, 6 2, 6],
+];
+
+*/
+
+const getDistanceBetweenNodes = (nodeOneCoordinates, nodeTwoCoordinates) => {
+    const [nodeOneXcoordinate, nodeOneYCoordinate] = nodeOneCoordinates;
+    const [nodeTwoXcoordinate, nodeTwoYCoordinate] = nodeTwoCoordinates;
+
+    const distance = Math.sqrt(
+        Math.pow(nodeTwoXcoordinate - nodeOneXcoordinate, 2) +
+            Math.pow(nodeTwoYCoordinate - nodeOneYCoordinate, 2),
+    );
+
+    return distance;
+};
+
+const areConnected = (nodeOne, nodeTwo) => {
+    const [nodeOneXcoordinate, nodeOneYCoordinate, nodeOneFruitNum, nodeOneLengthDistance] =
+        nodeOne;
+    const [nodeTwoXcoordinate, nodeTwoYCoordinate, nodeTwoFruitNum, nodeTwoLengthDistance] =
+        nodeTwo;
+
+    const distanceBetweenNodes = getDistanceBetweenNodes(
+        [nodeOneXcoordinate, nodeOneYCoordinate],
+        [nodeTwoXcoordinate, nodeTwoYCoordinate],
+    );
+
+    const smallestDistanceBetweenTwoNodes = Math.min(nodeOneLengthDistance, nodeTwoLengthDistance);
+    return distanceBetweenNodes <= smallestDistanceBetweenTwoNodes;
```

Comment:
This line is very confusing. `distanceBetweenNodes <= smallestDistanceetweenTwoNodes`? I don't really know what that means.

Each of your nodes has a vine length. Really what you're comparing is the distance between the centers of the trees and the total vine length. Your names should make this clearer.